pkg/webhook/validate.go

```diff
@@ -189,55 +189,62 @@ func dynamicValidateSriovNetworkNodePolicy(cr *sriovnetworkv1.SriovNetworkNodePo
 	for _, node := range nodeList.Items {
 		if cr.Selected(&node) {
 			nodesSelected = true
-			for _, ns := range nsList.Items {
-				if ns.GetName() == node.GetName() {
-					if ok, err := validatePolicyForNodeState(cr, &ns, &node); err != nil || !ok {
-						return false, err
-					}
-				}
-			}
-			// validate current policy against policies in API (may not be converted to SriovNetworkNodeState yet)
-			for _, np := range npList.Items {
-				if np.GetName() != cr.GetName() && np.Selected(&node) {
-					if ok, err := validatePolicyForNodePolicy(cr, &np); err != nil || !ok {
-						return false, err
-					}
-				}
+			err = validatePolicyForNodeStateAndPolicy(nsList, npList, node, cr)
+			if err != nil {
+				return false, err
 			}
 		}
 	}
 
 	if !nodesSelected {
 		return false, fmt.Errorf("no matched node is selected by the nodeSelector in CR %s", cr.GetName())
 	}
-	if !interfaceSelected {
-		return false, fmt.Errorf("no supported NIC is selected by the nicSelector in CR %s", cr.GetName())
-	}
 
 	return true, nil
 }
 
-func validatePolicyForNodeState(policy *sriovnetworkv1.SriovNetworkNodePolicy, state *sriovnetworkv1.SriovNetworkNodeState, node *corev1.Node) (bool, error) {
+func validatePolicyForNodeStateAndPolicy(nsList *sriovnetworkv1.SriovNetworkNodeStateList, npList *sriovnetworkv1.SriovNetworkNodePolicyList, node corev1.Node, cr *sriovnetworkv1.SriovNetworkNodePolicy) error {
```

Comment:
Shouldn't be pass node by pointer and not value? E.g. `node *corev1.Node` like in `validatePolicyForNodeState()`